backend/audit/etl.py

```diff
@@ -0,0 +1,312 @@
+        findings_uniform_guidance = (
+            self.single_audit_checklist.findings_uniform_guidance
+        )
+        findings_uniform_guidance_entries = (
+                findings_uniform_guidance
+                ["FindingsUniformGuidance"]
+                ["findings_uniform_guidance_entries"]
+            )
+
+        for entry in findings_uniform_guidance_entries:
+            findings = entry["findings"]
+            finding = Finding(
+                finding_ref_number=findings["reference_number"],
+                prior_finding_ref_numbers=findings.get("prior_references"),
+                modified_opinion=entry["modified_opinion"] == "Y",
+                other_non_compliance=entry["other_matters"] == "Y",
+                material_weakness=entry["material_weakness"] == "Y",
+                significant_deficiency=(
+                    entry["significant_deficiency"] == "Y"
+                ),
+                other_findings=entry["other_findings"] == "Y",
+                questioned_costs=entry["questioned_costs"] == "Y",
+                repeat_finding=(
+                    findings["repeat_prior_reference"] == "Y"
+                ),
+                type_requirement=(
+                    entry["program"]["compliance_requirement"]
+                ),
+                report_id=self.report_id,
+                award_id=entry.get("seq_number", 1), # TODO: This will be the sequence number
+                # audit_id=audit_id,
+                # audit_findings_id=audit_id,
+                # audit_year=self.audit_year,
+                # dbkey=None,
+                # is_public=self.is_public,
+            )
+            finding.save()
+    
+    def load_federal_award(self):
+        federal_awards = self.single_audit_checklist.federal_awards
+        for entry in federal_awards["FederalAwards"]["federal_awards"]:
+            program = entry["program"]
+            agency_cfda = (
+                f"{entry['federal_agency_prefix']}."
+                f"{entry['three_digit_extension']}"
+            )
+            loan = entry["loan_or_loan_guarantee"]
+            is_direct = (
+                entry["direct_or_indirect_award"]
+                ["is_direct"] == "Y"
+            )
+            is_passthrough = (
+                entry["subrecipients"]["is_passed"] == "Y"
+            )
+            cluster = entry["cluster"]
+            subrecipient_amount = (
+                entry["subrecipients"]
+                .get("subrecipient_amount")
+            )
+            loan_balance = (
+                loan.get("loan_balance_at_audit_period_end")
+            )
+            state_cluster_name = (
+                cluster.get("state_cluster_name")
+            )
+            other_cluster_name = (
+                cluster.get("other_cluster_name")
+            )
+            federal_award = FederalAward(
+                award_id=entry.get("seq_number", 1),
+                federal_program_name=program["program_name"],
```

Comment:
I think these are getting picked up now from an enumeration that is burried in the Jsonnet validation code. 

https://github.com/GSA-TTS/FAC/blob/main/backend/schemas/source/base/FederalProgramNames.json

specifically. That list will be considered authoritative going forward. The cluster names also get worked into the validation schemas.

Don't know if that helps.